File: pinyougou-interface/src/main/java/cn/itcast/core/service/AddressService.java
package cn.itcast.core.service;

import cn.itcast.core.pojo.address.Address;
import cn.itcast.core.pojo.address.Areas;
import cn.itcast.core.pojo.address.Cities;
import cn.itcast.core.pojo.address.Provinces;

import java.util.List;

public interface AddressService {
    List<Address> findListByLoginUser(String name);

    List<Provinces> findByProvinces();

    List<Cities> findByCitiesId(String parentId);

    List<Areas> findByAreasId(String parentId);

    Address findOne(Long id);

    void add(Address address);

    void update(Address address);
}

File: pinyougou-web-user/src/main/webapp/js/service/addressService.js
//服务层
app.service('addressService',function($http){

    this.findByProvinces = function(){
        return $http.get("../address/findByProvinces.do");
    }
    this.findByCitiesId = function(parentId){
        return $http.get("../address/findByCitiesId.do?parentId="+parentId);
    }
    this.findByAreasId = function(parentId){
        return $http.get("../address/findByAreasId.do?parentId="+parentId);
    }
    //获取当前登录账号的收货地址
    this.findAddressList=function(){
        return $http.get('../address/findListByLoginUser.do');
    }
    this.add = function(entity){
        return $http.post("../address/add.do",entity);
    }

    this.update=function(entity){
        return $http.post("../address/update.do",entity);
    }
    this.findOne=function(id){
        return $http.get("../address/findOne.do?id="+id);
    }
});
